grr/server/grr_response_server/gui/ui/components/expandable_hash/expandable_hash.ts

```diff
@@ -38,22 +15,57 @@ export function truncateIfNeeded(fullText: string): string {
 export class ExpandableHash {
   @Input() hashes?: Hash;
 
-  shouldDisplayPopup = false;
+  shouldShowPopup = false;
 
-  get truncatedSha256(): string {
-    if (this.hashes?.sha256) {
-      return truncateIfNeeded(this.hashes.sha256);
-    } else {
-      return SHA_256_NA_MESSAGE;
-    }
+  private mouseInsidePopup = false;
+  private mouseInsideTextArea = false;
+
+  private lastTimeout?: number;
+
+  get atLeastOneHashAvailable() {
+    return (this.hashes?.sha256 ?? this.hashes?.sha1 ?? this.hashes?.md5) !== undefined;
   }
 
-  constructor() { }
+  constructor(private changeDetectorRef: ChangeDetectorRef) { }
 
-  mouseEnteredTruncatedHash() {
-    this.shouldDisplayPopup = true;
+  mouseEnteredTextArea() {
+    this.showPopupNow();
+    this.mouseInsideTextArea = true;
+  }
+  mouseLeftTextArea() {
+    this.schedulePopupHiding();
+    this.mouseInsideTextArea = false;
+  }
+
+  mouseEnteredPopup() {
```

Comment:
Is there a way for Angular Material / CDK to handle when to show and hide the popup? Hardcoding the behavior and measuring MOUSE_IN_FLIGHT makes the code more complex than it needs to be and likely does not fully respect Material and Platform UX standards.

Reply:
I couldn't find one. Another option would be to position the pop-up exactly below where the mouse is (over the "Copy value" text). In this case we can just hide it as soon as the mouse leaves the pop-up, so that's simpler.

However, that's a bit tricky to implement, because the pop-up position could change if there's no space to display it at its usual position.

Any ideas?